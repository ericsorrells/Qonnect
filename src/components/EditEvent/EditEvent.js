import React from 'react';

const EditEvent = () => {
  return(
    <div>Edit Event Page :) !!!!</div>
  )
}

export default EditEvent;