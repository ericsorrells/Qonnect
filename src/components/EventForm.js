// ========================================================================================
import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
// ========================================================================================

class EventForm extends React.Component {
  constructor(props) {
    super(props)

    const { event } = props;
    this.state = {
      eventName:   event ? event.eventName    : '',
      date:        event ? event.date         : '',
      time:        event ? event.time         : '',
      location:    event ? event.location     : '',
      category:    event ? event.category     : '',
      imageUrl:    event ? event.imageUrl     : '',
      description: event ? event.description  : '',
      focused: ''
    }

    this.onFormSubmit        = this.onFormSubmit.bind(this);
    this.onNameChange        = this.onNameChange.bind(this);
    this.onTimeChange        = this.onTimeChange.bind(this);
    this.onLocationChange    = this.onLocationChange.bind(this);
    this.onCategoryChange    = this.onCategoryChange.bind(this);
    this.onImageUrlChange    = this.onImageUrlChange.bind(this); 
    this.onDescriptionChange = this.onDescriptionChange.bind(this);
  }

  onFormSubmit(e) {
    e.preventDefault();
    const { firstName, lastName } = this.props.profile;

    this.props.onSubmit({
      userName:    `${firstName} ${lastName}`,
      uid:         this.props.auth.uid,
      eventName:   this.state.eventName,
      date:        moment(this.state.date).valueOf(),
      time:        this.state.time,
      location:    this.state.location,
      category:    this.state.category,
      imageUrl:    this.state.imageUrl,
      description: this.state.description
    })
  }

  onNameChange(e) {
    this.setState({ eventName: e.target.value })
  }

  onTimeChange(e) {
    console.log('TIME: ', e.target.value);
    this.setState({ time: e.target.value  })
  }

  onLocationChange(e) {
    this.setState({ location: e.target.value })
  }

  onCategoryChange(e){
    this.setState({ category: e.target.value })
  }

  onImageUrlChange(e) {
    this.setState({ imageUrl: e.target.value })
  }

  onDescriptionChange(e) {
    this.setState({ description: e.target.value })
  }

  onFocusChange({ focused }) {
    this.setState(() => ({ calendarFocused: focused }))
  }

  render() {
    return (
      <div className='form'>
        <form onSubmit={this.onFormSubmit} >
          <label> Event Name: </label>
          <input
            type='text'
            onChange={this.onNameChange}
            value={this.state.name}
          />

          <label> Event Date: </label>
          <SingleDatePicker
            date={this.state.date}
            onDateChange={ date => this.setState({ date: date }) }
            focused={this.state.focused}
            onFocusChange={({ focused }) => this.setState({ focused })}
            numberOfMonths={1}
            isOutsideRange={() => false} 
          />

          <label>Time:</label>
          <input type='time' onChange={this.onTimeChange}/>
          <label> Event Location: </label>
          <input
            type='text'
            onChange={this.onLocationChange}
            value={this.state.location}
          />

          <label>Category:</label>          
          <select onChange={this.onCategoryChange} value={this.state.category ? this.state.category : ''}>
            <option disabled selected value=''>Select A Category</option>
            <option value="Sports">Sports</option>
            <option value="Entertainment">Entertainment</option>
            <option value="Food">Food</option>
            <option value="Music">Music</option>
            <option value="Miscellaneous">Miscellaneous</option>
          </select>

          <label>Image URL:</label>
          <input
            type='text'
            onChange={this.onImageUrlChange}
            value={this.state.imageUrl}
          />

          <label> Description: </label>
          <textarea
            onChange={this.onDescriptionChange}
            value={this.state.description}
          />

          <button type='submit'>Submit</button>
        </form>
      </div>
    )
  }
}

export default EventForm;