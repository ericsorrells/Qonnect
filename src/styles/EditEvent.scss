.edit-event {
	margin: 30px auto;
	display: flex;
	justify-content: center;
}

.border {
	border: 1px solid $dark-grey;
}