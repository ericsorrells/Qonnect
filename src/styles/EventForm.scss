.form {
  display: flex;
  padding-top: 10px;
  padding-right: 20px;
}

.SingleDatePicker,
.SingleDatePicker_1,
.form__element {
  width: 100%;
  margin-top: 5px;
  margin-bottom: 15px;
  border-radius: 5px;
  border: none;
  color: $dark-grey;
  background-color: white;
}

.form__select,
.form__input {
  height: 40px;
}

.form__textarea {
  height: 75px;
}

.button__purple {
  background-color: $main-color;
  width: 100%;
  color: white;
  padding: 10px 30px;
  border: none;
  border-radius: 5px;
  height: 40px;
}

/* DatePicker Styles
=========================================*/

.SingleDatePickerInput__withBorder {
  border: none;
  border-radius: 5px;
}

.DateInput_input__focused {
  border-bottom: none; // color of bottom bar when selected
}

.CalendarDay__selected_span {
  background: rgb(144, 130, 224);
  color: rgb(110, 3, 8);
}

.CalendarDay__hovered_span,
.CalendarDay__hovered_span:hover{
  background: $main-color-light;
  color: white;
}

.CalendarDay__selected,
.CalendarDay__selected:active,
.CalendarDay__selected:hover {
  background: $main-color;
  color: white;
  border: 1px solid $main-color;
}

.CalendarDay__selected_span:active,
.CalendarDay__selected_span:hover {
  border: 1px solid $dark-grey;
  border-color: 1px solid $dark-grey;
}