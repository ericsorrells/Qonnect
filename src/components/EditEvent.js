// ========================================================================================
import React from 'react';
// ========================================================================================
import EventForm from '../components/EventForm';
// ========================================================================================

// TODO: Can this be a stateless component???
class EditEvent extends React.Component {
  constructor(props) {
    super(props)
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(event) {
    const { id, eventName, date, time, location, description } = event;
    // TODO: add support for persisting edited data to Firebase
    this.props.editEvent(id, { ...event })
    this.props.history.push(`/show-event/${id}`); 
  }

  render() {
    return (
      <div>
        <EventForm
          event={this.props.event}
          onSubmit={this.onSubmit}
          profile={this.props.profile}
          auth={this.props.auth}
        />
      </div>
    )
  }
}

export default EditEvent;