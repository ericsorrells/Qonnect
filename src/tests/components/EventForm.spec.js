// ========================================================================================
import React from 'react';
import { shallow } from 'enzyme';
import 'jest-enzyme';
import moment from 'moment';
// ========================================================================================
import EditEvent from '../../components/EditEvent';
import EventForm from '../../components/EventForm';
// ========================================================================================
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile'
// ========================================================================================

describe('<EventForm />', () => {
  describe('rendering', () => {
    it('renders the correct initial state', () => {
      const component = shallow(<EventForm />)
      expect(component).toMatchSnapshot();
    });

    it('renders correctly with data', () => {
      const component = shallow(<EventForm event={basicEvent} onSubmit={jest.fn()} />)
      expect(component).toMatchSnapshot();
    });
  });

  describe('state', () => {
    let component;
    beforeEach(() => {
      component = shallow(<EventForm />);
    })

    it('has the correct initial state', () => {
      expect(component.state('eventName')).toBe('');
      expect(component.state('date')).toBe('');
      expect(component.state('time')).toBe('');
      expect(component.state('category')).toBe('');
      expect(component.state('imageUrl')).toBe('');
      expect(component.state('description')).toBe('');
      expect(component.state('focused')).toBe(undefined);
    });

    it('updates eventName correctly', () => {
      const value = 'New Description';
      component.find('#eventNameInput').simulate('change', { target: { value } });
      expect(component.state('eventName')).toBe(value);
    });

    it('updates date correctly', () => {
      const now = moment();
      component.find('#datePicker').prop('onDateChange')(now);
      expect(component.state('date')).toEqual(now);
    });

    it('updates time correctly', () => {
      const value = '12:59 PM';
      component.find('#timeInput').simulate('change', { target: { value } });
      expect(component.state('time')).toBe(value)
    });

    it('updates location correctly', () => {
      const value = '524 Bryn Mawr Ln';
      component.find('#locationInput').simulate('change', { target: { value } });
      expect(component.state('location')).toBe(value);
    });

    it('updates category correctly', () => {
      const value = 'Music';
      component.find('#categoryInput').simulate('change', { target: { value } });
      expect(component.state('category')).toBe(value);
    });

    it('updates imageUrl correctly', () => {
      const value = 'https://myImage.com';
      component.find('#imageUrlInput').simulate('change', { target: { value } });
      expect(component.state('imageUrl')).toBe(value);
    });

    it('updates description correctly', () => {
      const value = 'This will be a fun event!';
      component.find('#descriptionInput').simulate('change', { target: { value } });
      expect(component.state('description')).toBe(value);
    });

    it('updates calendar focus correctly', () => {
      component.find('#datePicker').prop('onFocusChange')({ focused: true });
      expect(component.state('focused')).toBe(true);
    });
  });

  it('handles onFormSubmit()', () => {
    const onSubmitSpy = jest.fn();
    const component = shallow(
      <EventForm
        event={basicEvent}
        onSubmit={onSubmitSpy}
        profile={basicProfile}
        auth={ { uid: '123abc' } }
      />
    )

    component.find('form').simulate('submit', {
      preventDefault: () => { }
    });

    expect(onSubmitSpy).toHaveBeenLastCalledWith({
      userName:          `${basicProfile.firstName} ${basicProfile.lastName}`,
      uid:               '123abc',
      eventName:         basicEvent.eventName,
      date:              moment(basicEvent.date).valueOf(),
      time:              basicEvent.time,
      location:          basicEvent.location,
      category:          basicEvent.category,
      imageUrl:          basicEvent.imageUrl,
      description:       basicEvent.description,
      selectedUser:      'none',
      interestedUsers:   'none',
      uninterestedUsers: 'none',
    });
  });
});