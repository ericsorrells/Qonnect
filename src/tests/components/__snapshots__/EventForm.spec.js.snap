// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventForm /> rendering renders correctly with data 1`] = `
<div
  className="form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
       Event Name: 
    </label>
    <input
      className="form__element form__input"
      id="eventNameInput"
      onChange={[Function]}
      placeholder="Give Your Event A Good Name"
      type="text"
      value="Cool Event Title"
    />
    <label>
       Event Date: 
    </label>
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date={345600000}
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      id="datePicker"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonth={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <label>
      Time:
    </label>
    <input
      className="form__element form__input"
      id="timeInput"
      onChange={[Function]}
      type="time"
      value="02:13"
    />
    <label>
       Event Location: 
    </label>
    <input
      className="form__element form__input"
      id="locationInput"
      onChange={[Function]}
      placeholder="Give The Address Of Your Event"
      type="text"
      value="123 Elm St."
    />
    <label>
      Category:
    </label>
    <select
      className="form__element form__select"
      id="categoryInput"
      onChange={[Function]}
      value="sports"
    >
      <option
        disabled={true}
        selected={true}
        value=""
      >
        Select A Category
      </option>
      <option
        value="Sports"
      >
        Sports
      </option>
      <option
        value="Entertainment"
      >
        Entertainment
      </option>
      <option
        value="Food"
      >
        Food
      </option>
      <option
        value="Music"
      >
        Music
      </option>
      <option
        value="Miscellaneous"
      >
        Miscellaneous
      </option>
    </select>
    <label>
      Image URL:
    </label>
    <input
      className="form__element form__input"
      id="imageUrlInput"
      onChange={[Function]}
      placeholder="Give Your Event A Nice Image"
      type="text"
      value="https://image.com"
    />
    <label>
       Description: 
    </label>
    <textarea
      className="form__element form__textarea"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Describe Your Event"
      value="Lorem ipsum text"
    />
    <button
      className="button__purple"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`<EventForm /> rendering renders the correct initial state 1`] = `
<div
  className="form"
>
  <form
    onSubmit={[Function]}
  >
    <label>
       Event Name: 
    </label>
    <input
      className="form__element form__input"
      id="eventNameInput"
      onChange={[Function]}
      placeholder="Give Your Event A Good Name"
      type="text"
      value=""
    />
    <label>
       Event Date: 
    </label>
    <withStyles(SingleDatePicker)
      anchorDirection="left"
      appendToBody={false}
      block={false}
      calendarInfoPosition="bottom"
      customCloseIcon={null}
      customInputIcon={null}
      date=""
      daySize={39}
      disableScroll={false}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      id="datePicker"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepFocusOnInput={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      noBorder={false}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsSelected": [Function],
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month.",
          "jumpToPrevMonth": "Move backward to switch to the previous month.",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
          "moveFocusByOneMonth": "Switch months.",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
          "openThisPanel": "Open this panel.",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field.",
          "selectFocusedDate": "Select the date in focus.",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      regular={false}
      renderCalendarInfo={null}
      renderDayContents={null}
      renderMonth={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      small={false}
      verticalHeight={null}
      verticalSpacing={22}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <label>
      Time:
    </label>
    <input
      className="form__element form__input"
      id="timeInput"
      onChange={[Function]}
      type="time"
      value=""
    />
    <label>
       Event Location: 
    </label>
    <input
      className="form__element form__input"
      id="locationInput"
      onChange={[Function]}
      placeholder="Give The Address Of Your Event"
      type="text"
      value=""
    />
    <label>
      Category:
    </label>
    <select
      className="form__element form__select"
      id="categoryInput"
      onChange={[Function]}
      value=""
    >
      <option
        disabled={true}
        selected={true}
        value=""
      >
        Select A Category
      </option>
      <option
        value="Sports"
      >
        Sports
      </option>
      <option
        value="Entertainment"
      >
        Entertainment
      </option>
      <option
        value="Food"
      >
        Food
      </option>
      <option
        value="Music"
      >
        Music
      </option>
      <option
        value="Miscellaneous"
      >
        Miscellaneous
      </option>
    </select>
    <label>
      Image URL:
    </label>
    <input
      className="form__element form__input"
      id="imageUrlInput"
      onChange={[Function]}
      placeholder="Give Your Event A Nice Image"
      type="text"
      value=""
    />
    <label>
       Description: 
    </label>
    <textarea
      className="form__element form__textarea"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Describe Your Event"
      value=""
    />
    <button
      className="button__purple"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
